[izniy] iP

izniy opened the pull request:
### BearBot

> "Your tasks, your rules." - BearBot Team 🧸

BearBot is your favourite bear companion that you manage your tasks efficiently. It is
- Simple and intuitive
- Fun and friendly
- **Fast** 🚀 and reliable

To use BearBot, follow these steps:
1. Download from [here](https://github.com/izniy/ip)
2. Run the application
3. Add your tasks
4. Let BearBot handle the rest for you! 😃

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java developer, here's the `main` method:

```
public static void main(String[] args) {
        new BearBot().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

izniy committed:
Level-0: Add initial chatbot greeting and exit

izniy committed:
Level-1: Add echo feature

izniy committed:
Level-2: new commands AddCommand and ListCommand

izniy committed:
Level-3: Mark As Done

izniy committed:
added three Task types: Todo, Event, Deadline

izniy committed:
A-TextUiTesting: tested todo, list, mark

izniy committed:
Level-5: added BearBotException, EmptyDescriptionException and TaskLimitException

izniy committed:
Level-6: added DeleteCommand

izniy committed:
Level 7: Task saving and loading

izniy committed:
Implemented Level 8: Added support for LocalDate
- Tasks now use LocalDate instead of String for dates
- Added validation for correct date formats
- Updated saving and loading to support yyyy-mm-dd format

izniy committed:
A-MoreOOP: Added getter methods for size and single tasks in TaskList to improve encapsulation

izniy committed:
A-Packages: Added Parser and Storage packages to existing packages

izniy committed:
misc changes to README

izniy committed:
Merge branch 'add-gradle-support'

izniy committed:
A-JUnit: Created TaskTest and TodoTest to test the Task and Todo classes

izniy committed:
A-JavaDoc: Added JavaDoc comments to all classes in command package, all classes in tasks package and Ui class

izniy committed:
Resolved merge conflicts for branch-A-CodingStandard

izniy committed:
Level-9: Added FindCommand feature

izniy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/bearbot/commands/AddCommand.java
#	src/main/java/bearbot/tasks/Event.java
#	src/main/java/bearbot/tasks/Task.java
#	src/main/java/bearbot/tasks/Todo.java

izniy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bearbot/tasks/TaskList.java

SahanaAvantika commented on src/main/java/bearbot/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+        String commandWord = words[0];
+
+        switch (commandWord) {
+            case "list":
```
constants in an enum should follow UPPER_CASE_SNAKE_CASE convention

izniy replied:
thank you for pointing out!

SahanaAvantika commented on src/main/java/bearbot/Ui.java:
```diff
@@ -0,0 +1,27 @@
+     * Displays a welcome message when the program starts.
+     */
+    public void showWelcomeMessage() {
+        System.out.println(" ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ฅ՞•ﻌ•՞ฅ ");
```
This is soo cute!

izniy committed:
Add GUI to the chatbot using JavaFX technology

dingZvel commented on docs/README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke User Guide
+# bearbot.BearBot User Guide
```
Should it be BearBot instead of bearbot.BearBot here?

dingZvel commented on src/main/java/bearbot/BearBot.java:
```diff
@@ -0,0 +1,116 @@
+package bearbot;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import bearbot.commands.Command;
+import bearbot.exceptions.BearBotException;
+import bearbot.parser.Parser;
+import bearbot.storage.Storage;
+import bearbot.tasks.TaskList;
+import bearbot.ui.Ui;
+import javafx.application.Platform;
```
Good that all classes to import are specified clearly!

dingZvel commented on src/main/java/bearbot/BearBot.java:
```diff
@@ -0,0 +1,116 @@
+ * Main logic for BearBot, handling task management and user commands.
+ */
+public class BearBot {
+    private static final String FILE_PATH = "./data/bearbot.txt"; // Path to store tasks
```
Does this follow the naming convention?

dingZvel commented on src/main/java/bearbot/BearBot.java:
```diff
@@ -0,0 +1,116 @@
+    public void run() {
+        ui.showWelcomeMessage();
+
+        String input;
```
Would it be better if you declare the input in the smallest scope and initialize it when it is declared?

dingZvel reviewed (commented):
LGTM! Just a few nits to fix.

WilkinsAng commented on src/main/java/bearbot/BearBot.java:
```diff
@@ -0,0 +1,116 @@
+        TaskList tempTaskList;
+        try {
+            tempTaskList = new TaskList(storage); // Load tasks from storage
+        } catch (Exception e) {
```
Would it be better to specify a smaller scope of Exception, rather than trying to "Catch them all", to be more inline with OOP principles?

WilkinsAng commented on src/main/java/bearbot/storage/Storage.java:
```diff
@@ -0,0 +1,55 @@
+
+import bearbot.tasks.Task;
+
+import java.io.*;
```
Perhaps you could split this up and import them explicitly?

WilkinsAng commented on src/main/java/bearbot/tasks/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+
+import bearbot.storage.Storage;
+
+import java.io.*;
```
This one as well. Perhaps you could import them explicitly.

WilkinsAng commented on src/main/java/bearbot/tasks/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+        saveTasks();
+    }
+
+    public List<Task> findTasks(String keyword) {
```
Maybe you could add javadocs for this as well?

WilkinsAng reviewed (commented):
I have notices some wildcard importing for you java.io.
But overall, your code is clean and have good coding conventions!
Jiayous for the rest of 2103!

izniy committed:
Add assertions for task index validation in DeleteCommand, MarkCommand, and UnmarkCommand

Task index validation relies on exceptions to handle out-of-bounds cases before modifying the task list.

Using exceptions for index validation treats predictable issues as runtime errors rather than enforcing expected constraints. Assertions are more appropriate for assumptions that should always hold in normal execution.

Use assert statements in the following classes to validate task indices before accessing the task list.
- DeleteCommand
- MarkCommand
- UnmarkCommand

Assertions explicitly document assumptions about valid task indices and prevent invalid operations during development without affecting runtime performance in production.

izniy committed:
Reduce nesting of the happy part of the code in FindCommand

izniy committed:
Merge pull request #1 from izniy/branch-A-Assertions

Add assertions for task index validation in DeleteCommand, MarkComman…

izniy committed:
Merge branch 'master' into branch-A-CodeQuality

izniy committed:
Merge pull request #2 from izniy/branch-A-CodeQuality

Reduce nesting of the happy part of the code in FindCommand

izniy committed:
Merge branch 'master' into branch-A-CodeQuality

izniy committed:
Sort out miscellaneous files

izniy committed:
Provide a way to archive items so that the user can remove items from the app but still keep a record of them somewhere

izniy committed:
Wrap up unfinished work before switching branches

izniy committed:
Merge branch 'branch-Level-10'

izniy committed:
Merge branch 'branch-A-CodeQuality'

izniy committed:
Change the colour of the user messages to make GUI more pleasing to look at

izniy committed:
Add pastel background to portray BearBot as a soothing companion

izniy committed:
Add cleanInput method to ensure unnecessary spaces do not break parsing

izniy committed:
Add DeadlineTest to ensure Deadline behaves in the intended way

izniy committed:
Add User Guide to provide the reader with enough guidance to use all important features of BearBot

izniy committed:
Edit User Guide to be more aesthetically pleasing for users

izniy committed:
Edit User Guide to ensure product screenshot is displayed

izniy committed:
Edit User Guide to ensure product screenshot is displayed

izniy committed:
Edit User Guide to ensure product screenshot is displayed

izniy committed:
Edit User Guide to ensure product screenshot is displayed properly

izniy committed:
Edit User Guide to ensure product screenshot is displayed

izniy committed:
Edit User Guide to ensure product screenshot is displayed properly

izniy committed:
Update README.md

izniy committed:
Update README.md

izniy committed:
Update README.md

izniy committed:
Edit User Guide to reflect updates

izniy committed:
Remove irrelevant comments

izniy committed:
Ensure correct starting file

izniy committed:
Edit dependencies

izniy committed:
Add application name

izniy committed:
Edit application name

izniy committed:
Edit application name

izniy committed:
Edit application name

izniy committed:
Edit product screenshot
